packages/interface/src/connection/index.ts

```diff
@@ -178,6 +178,12 @@ export interface NewStreamOptions extends AbortOptions {
    * for the protocol
    */
   maxOutboundStreams?: number
+
+  /**
+   * Opt-in to running over a limited connection - one that has time/data limits
+   * placed on it.
+   */
+  allowOnLimitedConnection?: boolean
```

Comment:
I think for consistency we should the terminology [transient connections](https://github.com/libp2p/go-libp2p/blob/260b9695cafdd8e35ec65b30ef153f0c15549c72/core/network/network.go#L115) to be consistent with other implementations

Reply:
I agree in principal, but the only [documentation](https://docs.libp2p.io/concepts/security/dos-mitigation/#transient-connections) I could find about transient connections in libp2p says:

1. A transient connection is a connection that has yet to have encryption or a muxer negotiated
2. go-libp2p calls this "transient", rust-libp2p calls this "negotiating"

So there doesn't seem to be much consistency across implementations and we're dealing with connections with data/time limits here which is different, hence the different name.

Reply:
Clarification on slack: data/time limited connections are "transient" in go-libp2p, but the resource manager also has a "transient" scope for newly opened streams that have yet to have encryption or a muxer negotiated.

🫠

Reply:
Renamed.